File: 27 Remove Element/27.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        if not nums or not len(nums):
            return len(nums)
        slow, fast = 0, 0
        while fast < len(nums):
            if nums[fast] == val:
                fast += 1
            else:
                nums[slow] = nums[fast]
                slow += 1
                fast += 1
        return slow

File: 27 Remove Element/27.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int slow = 0, fast = 0;
        while (fast < nums.length) {
            if (nums[fast] == val) {
                fast++;
            } else{
                nums[slow++] = nums[fast++];
            }
        }
        return slow;
    }
    
    // Time complexity: O(n)
    // Space complexity: O(1)
}